src/PrettyPrinting.jl

```diff
@@ -1401,4 +1403,158 @@ function with_unicode(f::Function)
   allow_unicode(old_allow_unicode);
 end
 
+################################################################################
+#
+#  IO context with indendation and lowercasing
```

Comment:
Is this code written from scratch or does it take inspiration from some existing code? In the latter perhaps we should acknowledge that?